mining.go

```diff
@@ -568,10 +589,62 @@ func createCoinbaseTx(subsidyCache *standalone.SubsidyCache, coinbaseScript []by
 	return dcrutil.NewTx(tx), nil
 }
 
+// createTreasuryBaseTx returns a coinbase transaction paying an appropriate subsidy
+// based on the passed block height to the treasury.
+func createTreasuryBaseTx(subsidyCache *standalone.SubsidyCache, opReturnPkScript []byte, nextBlockHeight int64, voters uint16, params *chaincfg.Params) (*dcrutil.Tx, error) {
+	tx := wire.NewMsgTx()
+	// All treasury base transactions require TxVersionTreasury.
+	tx.Version = wire.TxVersionTreasury
+
+	tx.AddTxIn(&wire.TxIn{
+		// Stakebase transactions have no inputs, so previous outpoint
+		// is zero hash and max index.
+		PreviousOutPoint: *wire.NewOutPoint(&chainhash.Hash{},
+			wire.MaxPrevOutIndex, wire.TxTreeRegular), // XXX should this not be wire.TxTreeStake?
```

Comment:
Nope.  There is no previous outpoint to refer to, so it should just be regular.  In fact, it should fail if it is not, and I know we recently addressed that.